packages/core/src/store.ts

```diff
@@ -1,34 +1,23 @@
-import thunk from 'redux-thunk';
-import { jsonformsReducer } from './reducers';
-import { applyMiddleware, createStore, Store } from 'redux';
-import { JsonForms } from './core';
-import { INIT, VALIDATE } from './actions';
+import { Store } from 'redux';
 import { JsonSchema } from './models/jsonSchema';
 import { UISchemaElement } from './models/uischema';
-import { generateDefaultUISchema, generateJsonSchema } from './generators';
 
-export const createJsonFormsStore = (initialState: JsonFormsState): JsonFormsStore => {
-  // TODO: typing
-  const store = createStore(
-    jsonformsReducer(),
-    initialState,
-    applyMiddleware(thunk)
-  );
-
-  return store as JsonFormsStore;
-};
 export interface JsonFormsStore extends Store<any> {
 }
 export interface JsonFormsState {
```

Comment:
can't we simply define the JSONForms sub state, so leave it as it was? What is the use case of a full JsonformsState?

Reply:
No sure if I understood the question correctly. Is it about the index properties?